packages/dashmate/src/listr/tasks/setup/regular/registerMasternodeGuideTaskFactory.js

```diff
@@ -0,0 +1,288 @@
+  or other external tools where keys are handled securely. During this process,
+  dashmate can optionally generate configuration elements as necessary, such as
+  the BLS operator key and the node id.\n`,
+              message: 'Which wallet will you use to store keys for your masternode?',
+              choices: [
+                { name: REGISTRARS.CORE, message: 'Dash Core Wallet' },
+                { name: REGISTRARS.ANDROID, message: 'Android Dash Wallet' },
+                { name: REGISTRARS.IOS, message: 'iOS Dash Wallet' },
+                { name: REGISTRARS.OTHER, message: 'Other' },
+              ],
+              initial: REGISTRARS.CORE,
+            },
+          ]);
+
+          let initialOperatorPrivateKey;
+          if (registrar === REGISTRARS.CORE || registrar === REGISTRARS.OTHER) {
+            ({ privateKey: initialOperatorPrivateKey } = await generateBlsKeys());
+          }
+
+          // TODO: We need to add description on how to find key generation form in the
+          //  specified wallet
+
+          const validateAddressWithNetwork = (value) => validateAddress(value, ctx.preset);
+
+          const collateralAmount = ctx.isHP === true
+            ? HPMN_COLLATERAL_AMOUNT
+            : MASTERNODE_COLLATERAL_AMOUNT;
+
+          const collateralDenomination = ctx.preset === PRESET_MAINNET ? 'DASH' : 'tDASH';
+
+          let state = {
+            collateral: {},
+            keys: {},
+            operator: {},
+            ipAndPorts: {},
+          };
+
+          let confirmation;
+          do {
+            const prompts = [
+              {
+                type: 'form',
+                name: 'collateral',
+                header: `  Dashmate needs to collect your collateral funding transaction hash and index.
+  The funding value must be exactly ${collateralAmount} ${collateralDenomination}.\n`,
+                message: 'Enter collateral funding transaction information:',
+                choices: [
+                  {
+                    name: 'txId',
+                    message: 'Transaction hash',
+                    validate: validateTxHex,
+                    initial: state.collateral.txId,
+                  },
+                  {
+                    name: 'outputIndex',
+                    message: 'Output index',
+                    validate: validatePositiveInteger,
+                    initial: state.collateral.outputIndex,
+                  },
+                ],
+                validate: ({ txId, outputIndex }) => validateTxHex(txId)
+                  && validatePositiveInteger(outputIndex),
+              },
+              {
+                type: 'form',
+                name: 'keys',
+                header: `  Dashmate needs to collect details about the owner, voting and payout addresses
+  to use in the masternode registration transaction. These are regular Dash
+  addresses, encoded in base58 format.\n`,
+                message: 'Enter DIP3 masternode addresses:',
```

Comment:
Honestly not sure we need to include DIP3 anymore. This has been the only type of masternode we've had for 3+ years now.
```suggestion
                message: 'Enter masternode addresses:',
```